crates/fuel-core/src/query/balance/asset_query.rs

```diff
@@ -140,12 +140,20 @@ impl<'a> AssetsQuery<'a> {
                     Ok(Resource::Message(message))
                 })
             })
-            .filter_ok(|message| {
-                if let Resource::Message(message) = message {
-                    message.fuel_block_spend.is_none()
-                } else {
-                    true
+            .filter_map(|message| match message {
+                Ok(message) => {
+                    if let Resource::Message(m) = &message {
+                        // Only return messages that have not been spent.
+                        match self.database.message_spent(&m.id()) {
```

Comment:
We should already fetch messages with this information. Could you add a new type, `CompressedMessage`, please?

`CompressedMessage` will be a message only with fields that are required by the blockchain. `Message` will be the entity used by GraphQL with `CoinStatus`. We already do the same for `Coin` and `CompressedCoin`.